### DISCUSSION TITLE
List of DAGMC tools

### DISCUSSION THREAD

**Discussion by shimwell** [svalinn/DAGMC#812] [Announcements]
We have a growing number of software tools and I thought it might be worth making a list of everything with a brief description.

| Name | Link | Description |
|---|---|---|
| DAGMC-stats | https://github.com/svalinn/dagmc_stats  | Tool for calculating and reporting statistics about DAGMC models  |
| Cubit-plugin | https://github.com/svalinn/Cubit-plugin  | Plugins and command extensions for Coreform Cubit |
| remove_dagmc_tags | https://github.com/svalinn/remove_dagmc_tags | A python package and command line tool for removing DAGMC tags such as graveyard and vacuum  |
| mcnp2cad | https://github.com/svalinn/mcnp2cad | Convert MCNP input files to a general CAD format  |
| vertices_to_h5m | https://github.com/fusion-energy/vertices_to_h5m | Converts mesh vertices and connectivity to h5m geometry files compatible with DAGMC simulations 
| cad_to_dagmc | https://github.com/fusion-energy/cad_to_dagmc | Convert CAD geometry (STP files) and Cadquery shapes to DAGMC h5m files  |
| brep_to_h5m | https://github.com/fusion-energy/brep_to_h5m | Converts Brep CAD geometry files to h5m geometry files compatible with DAGMC simulations  |
| cad_to_h5m | https://github.com/fusion-energy/cad_to_h5m | Converts CAD file(s) such as STP and SAT to a h5m file compatible with DAGMC based simulations using the Cubit Python API  |
| stl_to_h5m | https://github.com/fusion-energy/stl_to_h5m | Convert non overlapping STL files into a DAGMC h5m file complete with material tags and ready for use in neutronics simulations. |
| dagmc_geometry_slice_plotter | https://github.com/fusion-energy/dagmc_geometry_slice_plotter | A minimal Python package that produces slice plots through h5m DAGMC geometry files|
| brep_part_finder | https://github.com/fusion-energy/brep_part_finder | A Python package to identify the part ID number in Brep format CAD files 
| dagmc_h5m_file_inspector | https://github.com/fusion-energy/dagmc_h5m_file_inspector | Extracts information from DAGMC h5m files including volumes number, material tags |
| dagmc_bounding_box | https://github.com/fusion-energy/dagmc_bounding_box | Finds the bounding box and related properties of a DAGMC geometry  |
| dagmc_geometry_slice_plotter | https://github.com/fusion-energy/dagmc_geometry_slice_plotter | A minimal Python package that produces slice plots through h5m DAGMC geometry files  |
| CAD_to_openMC | https://github.com/openmsr/CAD_to_openMC | code to target the conversion from a step-file to a h5m-geometry for neutronics 
| OpenMC-plot | https://github.com/fusion-energy/openmc_plot | Plotting of OpenMC and DAGMC geometry 
| Stellarmesh | https://github.com/Thea-Energy/stellarmesh | Stellarmesh is a Gmsh wrapper and DAGMC geometry creator
| pydagmc |  https://github.com/svalinn/pydagmc | A convenience interface for examining and reassigning metadata in DAGMC models with PyMOAB

**Comment by jbae11** [comment]
I'm probably asking for too much, but it'd be awesome to have a VM or some sort of container that has all these tools built.. is this even possible?

**Comment by makeclean** [comment]
I don't see why not, we could certainly have one that just contains the DAGMC 'tools' that don't need specific monte carlo codes?
________________________________

**Comment by jbae11** [comment]
of course, and Cubit is also still EC?

**Reply by gonuke** [reply]
Cubit Learn is available without many restrictions, but prob can't be distributed in a VM

**Reply by shimwell** [reply]
You could try something like this for non commercial use cases
https://github.com/fusion-energy/cad_to_h5m/blob/d4a997204d8101acea16d4c70b86647566330eca/Dockerfile#L53

**Reply by bam241** [reply]
Have we checked with Cubit they are ok with this ?

**Reply by shimwell** [reply]
Yes I've checked a few times, see [this](https://github.com/fusion-energy/paramak/pull/64) thread for an example

**Comment by shimwell** [comment]
Just added Stellarmesh https://github.com/Thea-Energy/stellarmesh Stellarmesh is a Gmsh wrapper and DAGMC geometry creator

**Comment by pshriwise** [comment]
Submitting a request to the curator for [pydagmc](https://github.com/svalinn/pydagmc) 😄

**Reply by shimwell** [reply]
Added

**Comment by shimwell** [comment]
Perhaps we should make an awesome DAGMC repo as this list has grown quite a lot

**Reply by ahnaf-tahmid-chowdhury** [reply]
Similar to [openmc-echosystem](https://github.com/openmc-dev/openmc-ecosystem)? I am in.
